packages/assets/test/detections.tests.ts

```diff
@@ -1,31 +1,21 @@
 import { Assets, detectCompressedTextures } from '@pixi/assets';
-import { extensions } from '@pixi/core';
 
 describe('Detections', () =>
 {
     beforeEach(() => Assets.reset());
 
-    it('should have default detections', () =>
-    {
-        expect(Assets['_detections']).toHaveLength(2);
-    });
-
     it('should add compressed texture formats', async () =>
     {
-        extensions.add(detectCompressedTextures);
         await Assets.init();
         expect((Assets.resolver['_preferredOrder'][0].params.format as string[]).includes('s3tc')).toBe(true);
-        extensions.remove(detectCompressedTextures);
     });
 
     it('should remove any unsupported formats', async () =>
     {
-        extensions.add(detectCompressedTextures);
         detectCompressedTextures.test = jest.fn(async () => false);
         await Assets.init();
         expect(Assets.resolver['_preferredOrder'][0].params.format).toEqual(
-            ['avif', 'webp', 'png', 'jpg', 'jpeg']
+            ['webp', 'avif', 'png', 'jpg', 'jpeg']
```

Comment:
I think this highlights a problem. load/detection parsers are order specific sometimes.

For example, the detection parsers add themselves into the array in the order they are called, but now `webp` is being called before `avif`, meaning the priority has changed.

This could also be an issue for the loaders as you could have a generic `loadTxt` and `loadSpecificTxt` and you would want the specific version to be added after the generic one.

Maybe this means these shouldn't be extensions, or the extensions API needs a `addBefore`/`addAfter`, or a priority system?

Reply:
In v7 there already is a priority flag on ExtensionMetadata. I can fix that.

Reply:
I reworked this a little bit so that all the detections had a priority. I also created an explicit detection for populating the default formats, the unshift was doing some funny business.